Add GHCup-00210 (#458)
haskellfoundation/error-message-index: message-index/messages/GHCup-00210/index.md

---
title: GPG verify failed
summary: GHCup tried to verify the authenticity of a yaml file through GPG signatures, but failed
severity: error
introduced: 0.1.19.0
---

> GHCup was asked to verify all downloaded yaml files via GPG. E.g.

Reviewer: ```suggestion
GHCup receives information about available Haskell tools through a metadata file in YAML format that describes where to find the tools. As a defense against unauthorized third parties misleading GHCup into installing malicious software, GHCup supports checking that these metadata files are signed by a trusted [GPG](https://en.wikipedia.org/wiki/GNU_Privacy_Guard) key. This error occurs when GHCup was asked to verify all downloaded YAML files via GPG. E.g.
```

I think it's useful to provide a bit more context about where this error comes from, what the files involved are, and which role they're playing. This will also help users evaluate whether they want to turn off the strict checks temporarily or wait for someone else to resolve the situation.

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  286k  100  286k    0     0  2848k      0 --:--:-- --:--:-- --:--:-- 2862k
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   682  100   682    0     0  11278      0 --:--:-- --:--:-- --:--:-- 11366
[ Info  ] verifying signature of: /home/hasufell/.ghcup/cache/ghcup-0.0.7.yaml
[ Error ] [GHCup-00210] GPG verify failed: GPG verify failed: Process "gpg" with arguments ["--batch",
[ ...   ]                                                                     "--verify", "--quiet",
[ ...   ]                                                                     "--no-tty",
[ ...   ]                                                                     "/home/hasufell/.ghcup/cache/ghcup-0.0.7.yaml.sig",
[ ...   ]                                                                     "/home/hasufell/.ghcup/cache/ghcup-0.0.7.yaml"] failed with exit code 2.
```

> 

Reviewer: ```suggestion

## Troubleshooting
```

> Make sure the remote file exists. It's named the same as the yaml, except it ends with `.sig`. E.g.:

Reviewer: ```suggestion
First, make sure the remote file exists. It's name is the same as the YAML file, except it ends with `.sig`. For example:
```

* https://raw.githubusercontent.com/haskell/ghcup-metadata/master/ghcup-0.0.7.yaml
* https://raw.githubusercontent.com/haskell/ghcup-metadata/master/ghcup-0.0.7.yaml.sig

Then make sure you have the required GPG public keys: https://www.haskell.org/ghcup/guide/#gpg-verification

> Run ghcup with `-v` to see the exact gpg error, e.g.:

Reviewer: ```suggestion
Finally, run `ghcup` with `-v` to see the exact `gpg` error, which can be used to diagnose why verification is failing. For example, if the key has not been imported, GPG will fail with the following message:
```
